### PR TITLE
feat(levm): run hive tests with levm, generate and publish report

### PR DESCRIPTION
**Pull request by JereSalo**
**Motivation**

**Description**

- We now can execute multiple transactions in a block and commit them after execution of them all.
- It refactors `get_state_transitions_levm`, now in `account_updates` we'll have only the things that have changed from the account, so that we don't have much redundant information

- We build the client image with LEVM, we run hive tests and then we publish them
- Add to the Dockerfile the possibility to add build flags.
- Avoid panicking if setup wasn't completed successfully, as no tests will run in that scenario.

We can run tests with hive but the problem is that these tests execute a lot of blocks at the beginning and we usually fail at some block because of a difference in state, I tried debugging some tests and I fixed some things in other PRs but at some point I got stuck and decided that it was a better idea to leave it as is and merge this to main.

### CODE REVIEW

File: crates/blockchain/Cargo.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ path = "./blockchain.rs"
 
 [features]
 default = ["c-kzg"]
+default-levm = ["default", "levm"]
```
Comment (ilitteri) [inline crates/blockchain/Cargo.toml:31]: Let's name the target to `levm`
```suggestion
levm = ["default", "levm"]
```

File: crates/vm/Cargo.toml [hunk 2]
```diff
@@ -39,6 +39,7 @@ path = "./vm.rs"
 
 [features]
 default = ["c-kzg", "blst"]
+default-levm = ["default", "levm"]
```
Comment (ilitteri) [inline crates/vm/Cargo.toml:42]: Ditto

File: crates/vm/vm.rs [hunk 3]
```diff
@@ -103,56 +159,49 @@ cfg_if::cfg_if! {
...
-                let receipt = Receipt::new(
-                    transaction.tx_type(),
-                    matches!(result.result, TxResult::Success),
-                    cumulative_gas_used,
-                    result.logs,
-                );
-                receipts.push(receipt);
+            let mut receipts = Vec::new();
+            let mut cumulative_gas_used = 0;
+            let mut temporary_cache: CacheDB = HashMap::new();
```
Comment (ilitteri) [inline crates/vm/vm.rs:173]: I think `block_cache` or `block_execution_cache` are proper names for this as this.

File: crates/vm/vm.rs [hunk 4]
```diff
@@ -164,6 +213,7 @@ cfg_if::cfg_if! {
             tx: &Transaction,
             block_header: &BlockHeader,
             db: Arc<dyn LevmDatabase>,
+            temporary_cache: CacheDB
```
Comment (ilitteri) [inline crates/vm/vm.rs:216]: Ditto

**Review by ilitteri — COMMENTED** [review]
LGTM. Left some suggestions.

File: crates/blockchain/Cargo.toml (hunk 1 above)
Comment (JereSalo) [reply crates/blockchain/Cargo.toml:31]: There is already a levm target
`levm = ["ethrex-vm/levm"]`
Do you think I should just do: `levm = ["default", "ethrex-vm/levm"]`?

File: crates/vm/vm.rs (hunk 3 above)
Comment (JereSalo) [reply crates/vm/vm.rs]: Great idea! Done [here](https://github.com/lambdaclass/ethrex/pull/1546/commits/30eecfe3384719a639d5d610816b35737c098676)

File: crates/blockchain/Cargo.toml (hunk 1 above)
Comment (ilitteri) [reply crates/blockchain/Cargo.toml:31]: Yes!

Comment (JereSalo) [reply crates/blockchain/Cargo.toml]: [Done](https://github.com/lambdaclass/ethrex/pull/1546/commits/8fa56fdff2db8f4edc9faa3f39eca9025ab1e6ad)

**Pull request merged**